app/components/Form/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable import/no-duplicates */
+import clsx from 'clsx';
+import React from 'react';
+import ReactDatePicker, { ReactDatePickerProps } from 'react-datepicker';
+import enGB from 'date-fns/locale/en-GB';
+import { setHours, setMinutes } from 'date-fns';
+import Label from '~/components/Label';
+import { ClassName } from '~/utils/types';
+import { CommonInputProps } from '../common';
+import ErrorMessage from '../ErrorMessage';
+import { isDateEqual } from '~/utils/timeHelpers';
+
+import styles from './DatePicker.module.scss';
+
+interface Props
+    extends CommonInputProps,
+        ClassName,
+        Pick<
+            ReactDatePickerProps,
+            | 'readOnly'
+            | 'disabled'
+            | 'minDate'
+            | 'maxDate'
+            | 'minTime'
+            | 'maxTime'
+        > {
+    value: Date | undefined;
+    placeholder?: string;
+    onChange(v: Date): void;
+    onBlur?(): void;
+}
+
+/**
+ * Component for picking date and time. Is also available as part of a <Form />.
+ *
+ * @example
+ * const [v, s] = useState();
+ *
+ * <DatePicker value={v} onChange={s} />
+ */
+export default function DatePicker({
+    value,
+    onChange,
+    onBlur,
+    label,
+    placeholder = 'DD-MM-YYYY at HH:MM:SS',
+    isInvalid,
+    error,
+    className,
+    minDate,
+    maxDate,
+    minTime = value && minDate && isDateEqual(value, minDate)
+        ? minDate
+        : setHours(setMinutes(new Date(), 0), 0),
+    maxTime = value && maxDate && isDateEqual(value, maxDate)
+        ? maxDate
+        : setHours(setMinutes(new Date(), 59), 23),
+    ...props
+}: Props) {
+    return (
+        <div className={clsx(styles.root, className)}>
+            {label && <Label className="mB5">{label}</Label>}
+            <ReactDatePicker
+                selected={value}
+                onChange={onChange}
+                onBlur={onBlur}
+                showTimeSelect
+                timeCaption="Time"
+                timeFormat="HH:mm"
+                timeIntervals={15}
+                dateFormat="dd-MM-yyyy 'at' HH:mm:ss"
```

Comment:
When we display dates, we usually use `yyyy-MM-dd`, maybe it should also be like that here?

Reply:
I agree, they should definitely be aligned.. but doesn't it make more sense to display it with year last, as dates are normally displayed?

Reply:
Is ISO 8601 not the norm?

Reply:
is ISO 8601 not the norm?

Reply:
seems like it is 😅 I'll adapt.